### PR TITLE
feat: add basic LocalHdWallet functionality

### PR DESCRIPTION
**Pull request by dreamqip**
In this pull request, I've introduced the fundamental functionality for a LocalHdWallet. I've included a straightforward test script within the prefabs scene to demonstrate error and success states. In the future, LocalHdWallet could also support multiple accounts per mnemonic and set additional seed passwords to secure the wallet even more.

To make Nethereum's HdWallet work, I had to update the Nethereum package to the latest version. Also, there's a lot of junk left from copying the latest Nethereum package, we can remove it if necessary.

### CODE REVIEW

File: Assets/Thirdweb/Core/Scripts/Transaction.cs [hunk 1]
```diff
@@ -372,7 +372,7 @@ private async Task<string> SendGasless()
 
                 var postData = new RelayerRequest(request, signature, forwarderAddress);
 
-                using UnityWebRequest req = UnityWebRequest.Post(relayerUrl, "");
+                using UnityWebRequest req = UnityWebRequest.PostWwwForm(relayerUrl, "");
```
Comment (0xFirekeeper) [inline Assets/Thirdweb/Core/Scripts/Transaction.cs:375]: Please keep this as post to be backwards compatible with 2021

File: Assets/Thirdweb/Core/Scripts/TransactionManager.cs [hunk 2]
```diff
@@ -118,7 +118,7 @@ public static async Task<TransactionReceipt> ThirdwebWriteRawResult<TWFunction>(
 
                 var postData = new RelayerRequest(request, signature, forwarderAddress);
 
-                using UnityWebRequest req = UnityWebRequest.Post(relayerUrl, "");
+                using UnityWebRequest req = UnityWebRequest.PostWwwForm(relayerUrl, "");
```
Comment (0xFirekeeper) [inline Assets/Thirdweb/Core/Scripts/TransactionManager.cs:121]: Please keep this as post to be backwards compatible with 2021

File: Assets/Thirdweb/Core/Scripts/Wallets/ThirdwebLocalHdWallet.cs [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+        private Web3 _web3;
+        private readonly WalletProvider _provider;
+        private readonly WalletProvider _signerProvider;
+
+        public ThirdwebLocalHdWallet()
+        {
+            _account = null;
+            _web3 = null;
+            // I guess, we can use the local wallet provider here to avoid breaking conditions that rely on the local wallet provider
+            _provider = WalletProvider.LocalWallet;
```
Comment (0xFirekeeper) [inline Assets/Thirdweb/Core/Scripts/Wallets/ThirdwebLocalHdWallet.cs:20]: This should say WalletProvider.LocalHdWallet, we can update special wallet cases elsewhere

File: Assets/Thirdweb/Core/Scripts/Wallets/ThirdwebLocalHdWallet.cs [hunk 4]
```diff
@@ -0,0 +1,90 @@
...
+            _signerProvider = WalletProvider.LocalWallet;
```
Comment (0xFirekeeper) [inline Assets/Thirdweb/Core/Scripts/Wallets/ThirdwebLocalHdWallet.cs:21]: We can keep this as WalletProvider.LocalWallet, similar to the Paper integration

File: Assets/Thirdweb/Core/Scripts/Wallets/ThirdwebLocalHdWallet.cs [hunk 5]
```diff
@@ -0,0 +1,90 @@
...
+        }
+
+        public Task<string> Connect(WalletConnection walletConnection, string rpc)
+        {
+            // For now we're setting the seed password to null, however in the future we may want to allow the user to set a password along with their mnemonic
+            // Also, we're only using the first account in the wallet, but in the future we may want to allow the user to select which account they want to use
+            _account = new Nethereum.HdWallet.Wallet(walletConnection.mnemonic, null).GetAccount(0);
```
Comment (0xFirekeeper) [inline Assets/Thirdweb/Core/Scripts/Wallets/ThirdwebLocalHdWallet.cs:28]: What happens if no mnemonic is passed? Should be handled here and likely randomly generated?
Also the password can be walletConnection.password

File: Assets/Thirdweb/Examples/Scripts/Prefabs/Prefab_ConnectWallet.cs [hunk 6]
```diff
@@ -193,7 +194,7 @@ private void ValidateConnection(WalletProvider walletProvider)
                     _email = SupportedWalletsUI[WalletProvider.MagicLink].emailInput.text;
                 }
             }
-            else if (walletProvider == WalletProvider.LocalWallet)
+            else if (walletProvider is WalletProvider.LocalWallet or WalletProvider.LocalHdWallet)
```
Comment (0xFirekeeper) [inline Assets/Thirdweb/Examples/Scripts/Prefabs/Prefab_ConnectWallet.cs:197]: Prefer to be on the safe side here in case older unity versions don't support C# 9 syntax, if it doesn't break 2021 it should be ok

**Comment by 0xFirekeeper** [comment]
Hello @dreamqip first of all thanks for the PR! Added a few minor preliminary comments. If possible, could you also:
- Add it to `Prefab_ConnectWallet` prefab (inspector) so we can test the Scene_Prefabs functionality with LocalHdWallet
- Do not add additional Nethereum DLLs if possible, but feel free to update the existing ones, assuming this is 4.16?

For edge cases related to LocalWallet like going through the TransactionManager, I can handle that at the end and make sure everything works.

Thanks again!

File: Assets/Thirdweb/Core/Scripts/Wallets/ThirdwebLocalHdWallet.cs (hunk 4 above)
Comment (dreamqip) [reply Assets/Thirdweb/Core/Scripts/Wallets/ThirdwebLocalHdWallet.cs:20]: Thanks for the tip, I've changed the provider to `WalletProvider.LocalHdWallet` as it supposed to be.

File: Assets/Thirdweb/Core/Scripts/TransactionManager.cs (hunk 2 above)
Comment (dreamqip) [reply Assets/Thirdweb/Core/Scripts/TransactionManager.cs:121]: Oh, I see. I'll do it right away

File: Assets/Thirdweb/Core/Scripts/Wallets/ThirdwebLocalHdWallet.cs (hunk 5 above)
Comment (dreamqip) [reply Assets/Thirdweb/Core/Scripts/Wallets/ThirdwebLocalHdWallet.cs:27]: If no mnemonic is passed, the user will get an `ArgumentNullException: Value cannot be null.` error. in the case of the empty string:  `NotSupportedException: Unknown`. Theoretically, we could randomly generate a mnemonic and then print it to the console, but I'm not sure that's the best option. If we want to pass `walletConnection.password` as the seed password, I strongly suggest we document it well. I had such a case where, in setting the password for the wallet, I passed the user's password to this option, and as a result, even if the user provided the correct mnemonic but a different password, an entirely new wallet would be created.

Comment (0xFirekeeper) [reply Assets/Thirdweb/Core/Scripts/Wallets/ThirdwebLocalHdWallet.cs:27]: I was thinking more for the situation where you just want to generate an HD Wallet, if you were building a wallet app that might be useful, specially in the future when we support multiple accounts.
As for password right now with `LocalWallet` if you don't pass it we use `SystemInfo.deviceUniqueIdentifier`, but either way works for me, not supporting it is fine in this context as well.

**Comment by 0xFirekeeper** [comment]
Hey @dreamqip - will be closing this for now and revisiting; it will be part of our upcoming EmbeddedWallet modular product, where LocalWallet will move into that option and you'll have a clear separation of auth methods and separation methods. So in this case, you'd be able to choose the strategy you want to encrypt your local wallet with. Apologies for not getting on this sooner, but longer term vision here will surely encapsulate this!

**Comment by dreamqip** [comment]
Great news! Sorry, that I couldn't finish the PR
